packages/vue/src/router.ts

```diff
@@ -72,76 +70,105 @@ export function vueRouterInstrumentation(
+    instrumentVueRouter(
+      router,
+      {
+        routeLabel: options.routeLabel || 'name',
+        instrumentNavigation: startTransactionOnLocationChange,
+        instrumentPageLoad: startTransactionOnPageLoad,
+      },
+      startTransaction,
+    );
+  };
+}
+
+/**
+ * Instrument the Vue router to create navigation spans.
+ */
+export function instrumentVueRouter(
+  router: VueRouter,
+  options: {
+    routeLabel: 'name' | 'path';
+    instrumentPageLoad: boolean;
+    instrumentNavigation: boolean;
+  },
+  startNavigationSpanFn: (context: TransactionContext) => void,
+): void {
+  router.onError(error => captureException(error, { mechanism: { handled: false } }));
+
+  router.beforeEach((to, from, next) => {
+    // According to docs we could use `from === VueRouter.START_LOCATION` but I couldnt get it working for Vue 2
+    // https://router.vuejs.org/api/#router-start-location
+    // https://next.router.vuejs.org/api/#start-location
+
+    // from.name:
+    // - Vue 2: null
+    // - Vue 3: undefined
+    // hence only '==' instead of '===', because `undefined == null` evaluates to `true`
+    const isPageLoadNavigation = from.name == null && from.matched.length === 0;
+
+    const attributes: SpanAttributes = {
+      [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.navigation.vue',
+      'routing.instrumentation': 'vue-router',
+    };
+
+    for (const key of Object.keys(to.params)) {
+      attributes[`params.${key}`] = to.params[key];
+    }
+    for (const key of Object.keys(to.query)) {
+      const value = to.query[key];
+      if (value) {
+        attributes[`query.${key}`] = value;
       }
+    }
 
-      if (startTransactionOnPageLoad && isPageLoadNavigation) {
-        // eslint-disable-next-line deprecation/deprecation
-        const pageloadTransaction = getActiveTransaction();
-        if (pageloadTransaction) {
-          const attributes = spanToJSON(pageloadTransaction).data || {};
-          if (attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE] !== 'custom') {
-            pageloadTransaction.updateName(transactionName);
-            pageloadTransaction.setAttribute(SEMANTIC_ATTRIBUTE_SENTRY_SOURCE, transactionSource);
-          }
-          // TODO: We need to flatten these to make them attributes
-          // eslint-disable-next-line deprecation/deprecation
-          pageloadTransaction.setData('params', data.params);
-          // eslint-disable-next-line deprecation/deprecation
-          pageloadTransaction.setData('query', data.query);
+    // Determine a name for the routing transaction and where that name came from
+    let transactionName: string = to.path;
+    let transactionSource: TransactionSource = 'url';
+    if (to.name && options.routeLabel !== 'path') {
+      transactionName = to.name.toString();
+      transactionSource = 'custom';
+    } else if (to.matched[0] && to.matched[0].path) {
+      transactionName = to.matched[0].path;
+      transactionSource = 'route';
+    }
```

Comment:
I think something here is slightly off but it's also off in the original implementation. See https://github.com/getsentry/sentry-javascript/pull/10476#discussion_r1476191552

Should we just always send `route`? I mean even if we take the name instead of the path, it still represents a route ultimately. Just a thought. Maybe we do it in a separate PR to fix this atomically in case we need to revert.

Reply:
yeah, good point 🤔 do we have any place where we rely on some behavior that if `source===route` we expect it to be a literal route, as in a URL path? 🤔 if not, probably OK to keep this `route` always I think - cc @AbhiPrasad

Reply:
afaik the only important distinction in Relay is between source `url` (where we apply server side clustering/grouping) vs all other sources (like `route` and `custom`). But not 100% sure anymore.

Reply:
I asked the ingest team if there's a difference.

Reply:
I guess internally we also only attach transaction names if they are not URLs to baggage.

I don't know the Relay specifics either, it'll be good to double check them.

Reply:
I'll merge this as is (as it is the same behavior as before) and as Lukas said, we can change it in a separate PR where we can be clear about the reasoning etc!